File: src/test/java/com/twsg/lnl/kotlin/c3_extensions/g1_java/CharacterUtil.java
package com.twsg.lnl.kotlin.c3_extensions.g1_java;

/**
 * @author tonyl
 */
public class CharacterUtil {

    private CharacterUtil() {
    }

    public static boolean isVowel(char ch) {
        switch (Character.toLowerCase(ch)) {
            case 'a':
            case 'e':
            case 'u':
            case 'i':
            case 'o':
                return true;
            default:
                return false;
        }
    }
}

File: src/test/java/com/twsg/lnl/kotlin/c3_extensions/g2_kotlin/CharacterUtil.kt
package com.twsg.lnl.kotlin.c3_extensions.g2_kotlin

/**
 * @author tonyl
 */

/* extension (computed) property */

val Char.isVowel
    get() = when (Character.toLowerCase(this)) {
        'a', 'e', 'u', 'i', 'o' -> true
        else -> false
    }
